src/Cardano/CLI/Query.hs

```diff
@@ -55,27 +57,48 @@ import qualified Data.Map.Strict as M
 import qualified Cardano.API.Jormungandr as Jormungandr
 
 data MetadataRetrievalError
```

Comment:
Since TxId is common to every constructor, you could factor it out.

```haskell
data MetadataRetrievalError {
  retrievalErrorTxId :: TxId
  retrievalErrorReason :: MetadataRetrievalErrorReason
}
```